unified-runtime/test/CMakeLists.txt

```diff
@@ -10,15 +10,91 @@ FetchContent_Declare(
   GIT_TAG        release-1.12.1
 )
 
+include(FindLit)
+
 set(UR_TEST_DEVICES_COUNT 1 CACHE STRING "Count of devices on which conformance and adapters tests will be run")
 set(UR_TEST_PLATFORMS_COUNT 1 CACHE STRING "Count of platforms on which conformance and adapters tests will be run")
 set(UR_TEST_FUZZTESTS ON CACHE BOOL "Run fuzz tests if using clang and UR_DPCXX is specified")
 # For Windows: Prevent overriding the parent project's compiler/linker settings
 set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
+
+if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang" AND UR_DPCXX AND UR_TEST_FUZZTESTS)
+    set(UR_FUZZTESTING_ENABLED ON)
+endif()
+
 set(INSTALL_GTEST OFF)
 FetchContent_MakeAvailable(googletest)
 enable_testing()
 
+# Set up the root `check-unified-runtime` target
+configure_file(lit.site.cfg.py.in lit.site.cfg.py)
+add_custom_target(deps_check-unified-runtime)
+if(UR_STANDALONE_BUILD)
+  add_custom_target(check-unified-runtime
+    COMMAND "${URLIT_LIT_BINARY}" "${CMAKE_CURRENT_BINARY_DIR}" -v
+    DEPENDS deps_check-unified-runtime
+  )
+else()
+  add_lit_testsuite(check-unified-runtime "Running Unified Runtime ${suite} tests"
+    ${CMAKE_CURRENT_BINARY_DIR}
+    DEPENDS FileCheck deps_check-unified-runtime
+  )
+endif()
+if(UR_STANDALONE_BUILD)
+  add_test(NAME lit COMMAND ${URLIT_LIT_BINARY} "${CMAKE_CURRENT_BINARY_DIR}")
+endif()
+
+# Create a new test suite that can be ran with `check-unified-runtime-${suite}`
+# Lit will execute tests in `${CMAKE_CURRENT_BINARY_DIR}`, and any `add_gtest_test`s will be added as dependencies of
+# this test suite
+function(with_ur_lit_testsuite suite)
```

Comment:
I find `with_ur_lit_testsuite` a confusing name for what this is actually doing, `add_ur_list_testsuite` would be much more intuitive.

Reply:
I was imagining something similar to Python's syntax, that it would read something like:

```
with ur_lit_testsuite():
  add_test()
  add_test()
```

I've renamed it to just `ur_lit_testsuite`, which is maybe closer to cmake terminology.

Reply:
`add_ur_lit_testsuite` is closer to `add_test`...

Reply:
It's not similar to `add_test` though. It sets up a "context" so that future `add_gtest_test` calls are attached to that test suite. Not sure the best way of naming things to make that clear.